packages/block-editor/src/components/inserter/block-list.js

```diff
@@ -168,22 +168,13 @@ function InserterBlockList( {
 
 	// Announce search results on change
 	useEffect( () => {
-		const resultCount = Object.keys( itemsPerCategory ).reduce(
-			( accumulator, currentCategorySlug ) => {
-				return (
-					accumulator + itemsPerCategory[ currentCategorySlug ].length
-				);
-			},
-			0
-		);
-
 		const resultsFoundMessage = sprintf(
 			/* translators: %d: number of results. */
-			_n( '%d result found.', '%d results found.', resultCount ),
-			resultCount
+			_n( '%d result found.', '%d results found.', filteredItems.length ),
+			filteredItems.length
 		);
 		debouncedSpeak( resultsFoundMessage );
-	}, [ itemsPerCategory, debouncedSpeak ] );
+	}, [ filteredItems.length, debouncedSpeak ] );
```

Comment:
An observation here: The previous behavior may or may not have been ideal, in that it could cause a new spoken message even when the actual number of results hasn't changed. This new implementation will only cause a new spoken message if the filtered count actually changes. It's not entirely clear to me which should be desirable though, since perhaps one might expect the spoken announcement in response to any change in the search query, regardless of whether it results in the same count of filtered items. In which case, I think the dependency here would either be `filteredItems` (the reference array of filtered items), or perhaps even `filterValue` itself (the search term).

Reply:
Users would expect the search message to be announced for any search. We need to take into consideration this is the only immediate feedback screen reader users will get for a search so, say they initially search for `xyz` and get "5 results found". Then they do a new search which coincidentally produces the same number of result and... they would just hear a resounding silence. No feedback.

Reply:
Thanks for the clarification @afercia . That makes sense to me.

In which case, this should probably be one or the other of `filteredItems` or `filterValue`. The former may be more "correct" technically speaking as the value referenced within the hook. The latter is more "correct" practically as being the trigger for a (debounced) spoken message.

Reply:
Updated in 32c3931 . I chose to use `filterValue` as the dependency, since the `filteredItems` value can change in response to underlying state changes not initiated by the user ([source](https://github.com/WordPress/gutenberg/blob/903e52c23ac9fea0b53b37625af6a9c60a11a704/packages/block-editor/src/components/inserter/block-list.js#L119), e.g. if block categories change for some reason). This would seem to me as an unexpected trigger for making an announcement. Welcome to further feedback, though.